refactor(*): bundle styles with components

simeonoff opened the pull request:
Closes #13770, #16794

### Additional information (check all that apply):
 - [ ] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [ ] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

simeonoff committed:
initial commit

simeonoff committed:
refactor(*): test scoping with container queries

simeonoff committed:
refactor(avatar): update styles

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
refactor(checkbox): bundle styles with component - initial commit

simeonoff committed:
refactor(checkbox): fix component styles

simeonoff committed:
build(styles): don't rely on theme service for bundled styles

simeonoff committed:
refactor(styles): add themed component styles to a layer

simeonoff committed:
demos(theme): update

simeonoff committed:
refactor(avatar): update base theme

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
fix(avatar): icon size

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
build(lig): build component styles before library

SisIvanova committed:
refactor(icon): bundle styles with component (#15281)

simeonoff committed:
fix(checkbox): make light themes functional again

simeonoff committed:
fix(checkbox): dark themes

simeonoff committed:
refactor(checkbox): untagle placeholder selectors

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
npm(package-lock): revert changes

simeonoff committed:
npm(package-lock): revert changes

simeonoff committed:
npm(package.json): revert file order

simeonoff committed:
build(npm): add back concurrently

simeonoff committed:
deps(node-watch): add back node-watch

simeonoff committed:
chore(*): remove some changes

simeonoff committed:
build(scripts): global themes not building

simeonoff committed:
fix(checkbox): wrong tickmark in both indeterminate and checked

simeonoff committed:
refactor(checkbox): organize variables better

simeonoff committed:
refactor(demos): revert back to material

simeonoff committed:
refactor(divider): convert to component and bundle styles (#15338)

SisIvanova committed:
refactor(badge): bundle styles with component (#15278)

simeonoff committed:
fix(badge): icon size

simeonoff committed:
refactor(badge): add layer mixin and fix icon size

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

SisIvanova committed:
refactor(dialog): bundle styles with component (#15341)

simeonoff committed:
fix(checkbox): label color fixes

simeonoff committed:
refactor(toast): bundle styles with component (#15344)

SisIvanova committed:
refactor(radio): bundle styles with component (#15294)

simeonoff committed:
refactor(themes): prefix layers with ig to avoid collisions

simeonoff committed:
refactor(checkbox): prevent FOUC

simeonoff committed:
refactor(core): add CSS property and animation to help with FOUC

SisIvanova committed:
refactor(navbar): bundle styles with component (#15345)

simeonoff committed:
refactor(snackbar): bundle styles with component (#15346)

SisIvanova committed:
refactor(switch): bundle styles with component (#15269)

SisIvanova committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
refactor(list): bundle styles with component (#15361)

simeonoff committed:
refactor(switch): fix thumb animation in safari

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
fix(checkbox): dark bootstrap uses wrong schema

simeonoff committed:
refactor(chip): bundle styles with component (#15371)

SisIvanova committed:
refactor(navdrawer): bundle styles with component (#15365)

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
fix(*): timepicker and navdrawer sample after merging master

simeonoff committed:
deps(theming): update to latest version

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
refactor(list): add back selected styles after merging master

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
refactor(card): bundle styles with component (#15450)

simeonoff committed:
refactor(banner): bundle styles with component (#15467)

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

Commit:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
refactor(bottom-nav): bundle styles with component (#15520)

simeonoff committed:
refactor(splitter): bundle styles with component (#15514)

simeonoff committed:
refactor(expansion-panel): bundle styles with component (#15476)

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
refactor(typography): put typography vars in a layer

SisIvanova committed:
Merge branch 'master' into simeonoff/scoped-styles

desig9stein committed:
refactor(stepper): scope styles to component (#15503)

SisIvanova committed:
refactor(carousel): bundle styles with component (#15473)

SisIvanova committed:
refactor(slider): bundle styles with component (#15558)

desig9stein committed:
[Tree]: scope the theme styles to the component. (#15565)

* refactor(tree): scope tree-theme styles to the component.

Remove tree typography and theme logic, add tree styles.

Commented out tree typography and theme generation across styles. Added new tree and tree-node styling files, including themes for material, indigo, bootstrap, and fluent. Integrated ViewEncapsulation and updated tree component to utilize new styles.

* Update projects/igniteui-angular/src/lib/tree/tree-node/themes/_base.scss

Co-authored-by: Dilyana Yarabanova <45598235+didimmova@users.noreply.github.com>

---------

Co-authored-by: Dilyana Yarabanova <45598235+didimmova@users.noreply.github.com>

SisIvanova committed:
refactor(switch): remove invalid state properties (#15610)

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
style(demos): card sample code formatting in template

SisIvanova committed:
refactor(btn-group):  bundle styles with component (#15569)

SisIvanova committed:
refactor(action-strip): bundle styles with component (#15618)

SisIvanova committed:
refactor(rating): bundle styles with component (#15641)

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

desig9stein committed:
refactor(tabs): scope styles to component (#15659)

SisIvanova committed:
refactor(paginator): bundle styles with component (#15655)

SisIvanova committed:
refactor(drop-down): bundle styles with component (#15630)

SisIvanova committed:
refactor(chip): apply latest bootstrap changes (#15677)

SisIvanova committed:
fix(navbar): nested components sizes (#15685)

SisIvanova committed:
refactor(grid-summaries): bundle styles with component (#15669)

SisIvanova committed:
fix(card): action-color not working (#15698)

simeonoff committed:
feat(standalone): add map splitting function

desig9stein committed:
refactor(calendar): scope themes to the component (#15373)

SisIvanova committed:
refactor(grid-toolbar): bundle styles with component (#15676)

SisIvanova committed:
fix(snackbar): button color

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
fix(samples): after upgrading to igniteui-webcomponents to version 6.0.0

desig9stein committed:
[Progress bars]: Bundle theme styles with component (#15720)

SisIvanova committed:
refactor(pivot-grid): bundle styles with component (#15772)

SisIvanova committed:
refactor(directives): wrap styles in base layer (#15759)

SisIvanova committed:
refactor(column-actions): bundle styles with component (#15765)

SisIvanova committed:
feat(radio-group): label styles (#15793)

SisIvanova committed:
refactor(ESF): update move and sort icon sizes

SisIvanova committed:
refactor(grid-summary): remove redundant code and colors (#15828)

* refactor(grid-summary): remove redundant code and colors

* deps(*): update theming version

desig9stein committed:
[input-group]: scope theme styles (#15782)

* refactor(input-group): initial themes structure,
Also adding styles for material box(input,textarea,select and combo).

* refactor(input-group): adding the rest of the material types styles

- Removing the logic for themes inside the .ts file, removing all classes related to specific themes.
- Updating the showcase sample
- wrapping the label theme in base layer

* refactor(input-group): adding styles for indigo

- modify the input group template
- disabled typography
- fix some layout issues
- move more styles to base
- update wc version and add file type input in the sample

* refactor(input-group): initial bootstrap styling

* refactor(input-group): finalize input group styling

* refactor(input-group): fix showcase sample error

* refactor(combo): prepare combo theme

* refactor(select): prepare select theme

* refactor(input-group): fix some layout issues
Add clear icon colors in file type input.

* refactor(input-group): finalize the input-group styles
also update the showcase sample to add remove the hint

* refactor(combo): finalize styles

* fix(input-group): use the correct color variable for focus background in indigo theme

* fix(input-group): revert the variable until we have an agreement on this

* fix(input-group): make sure the the upload button is using its own padding not the prefix/ suffix padding+

* fix(Combo): add display contents to the new wrapper around the combo search and content use it to only scope the variables

* fix(input-group): Make sure that the search use the same shadow on focus as it uses on hover

* chore(combo): cleanup commented code

* fix(input-group): mace sure that the default size is working as before

* fix(combo): dont remove the old class and fix the padding for the add container

* fix(input-group): remove untalented import from the showcase sample

* fix(combo): fix test for combo and simple combo

* fix(input-group): fix native input padding in suffixed/prefixed variants

* fix(combo): fix case sensitive icon color in fluent theme

* fix(combo): fix case sensitive icon color/border in bootstrap theme

* fix(combo): fix indigo caret color

* fix(input-group): rever back to hardcoded dolor for case-icon (required by the design team)

desig9stein committed:
refactor(input-group, textarea): label position

handle placeholder with css

SisIvanova committed:
fix(combo/select): remove suffix dynamically (#15961)

SisIvanova committed:
refactor(drop-down): items icon size (#15968)

desig9stein committed:
refactor(input-group): refactor file type input structure

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
fix(theme-token): match implementation in master

simeonoff committed:
deps(theming): bump to latest version

desig9stein committed:
fix(input-group): fix the showcase sample

desig9stein committed:
chore(igniteui-webcomponents): bump to v6.1.1

desig9stein committed:
fix(input-group): use the file schema colors

adrianptrv committed:
fix(grid & combo): remove grid column actions negative margin & fix combo clear button parameters (#15966)

* fix(grid & combo): remove grid negative margin & fix combo clear button parameters

* fix(combo): fix spacing

desig9stein committed:
refactor(input-group, textarea): label position (#15953)

handle placeholder with css

SisIvanova committed:
fix(radio-group): dynamically added radio buttons do not initialize (#16042)

SisIvanova committed:
fix(date-picker): indigo theme elevations (#16085)

adrianptrv committed:
fix(textarea): suffix space (#16140)

adrianptrv committed:
fix(input-group): prefix & suffix bootstrap border (#16151)

didimmova committed:
feat(combo): fix combo initial dropdown height (#16171)

* feat(combo): fix combo initial dropdown height

* refactor(themes): update dropdown theme to use a CSS variable for item count

---------

Co-authored-by: Simeon Simeonoff <sim.simeonoff@gmail.com>

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
fix(demos): not running after merging master

simeonoff committed:
lint(stylelint): fix errors

simeonoff committed:
spec(*): tests not running

desig9stein committed:
fix(input-group): fix broken input group after the merge conflict

desig9stein committed:
fix(button-group): remove unnecessary bootstrap-specific styles (#16260)

- We should not remove a border or any schema-connected style from the theme. Rather, we should update the variable in the schema, which in this case is already done.

desig9stein committed:
fix(card): make sure that actions text color is not overriding the button themes (#16265)

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
fix(tooltip): post-merge theme fixes

adrianptrv committed:
feat(checkbox): set tick-width property - (scoped) (#16321)

* feat(checkbox): set tick-width property

* chore(theming): bump version to 22.0.0-beta.1

* deps(theming): bump to latest version

---------

Co-authored-by: Simeon Simeonoff <sim.simeonoff@gmail.com>

didimmova committed:
feat(timepicker): update timepicker sample with properties panel (#16340)

SisIvanova committed:
refactor(input-group): add readonly styles (#16328)

* refactor(input-group): add readonly styles

* fix(input-group): material border styles

SisIvanova committed:
refactor(input-group): input refinement (#16307)

* refactor(input-group): input refinement

* refactor(input-group): address review comments

* fix(input-group): helper text issue

* Update _core.scss

---------

Co-authored-by: Simeon Simeonoff <sim.simeonoff@gmail.com>

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
refactor(themes): remove igx-file-input from base styles

simeonoff committed:
deps: bump to latest version of theming

simeonoff committed:
refactor(base): disable global time-picker theme

didimmova committed:
fix(checkbox): fix checkbox styles to be identic between projects (#16372)

* fix(checkbox): fix checkbox styles to be identic between projects

* fix(checkbox): fix latest addressed problems

SisIvanova committed:
feat(badge): design enhancement (#16396)

SisIvanova committed:
refactor(time-picker): bundle styles with component (#16418)

SisIvanova committed:
fix(select): style component using sass theme (#16404)

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

desig9stein committed:
fix(navbar): adjust icon sizes and spacing across themes (#16436)

- Standardize `--component-size` for icons in navbar themes.
- Remove redundant `--ig-size` definitions from shared themes.
- Add padding and auto sizing for indigo theme icons.
- Update icon button styles for consistent behavior across themes.

adrianptrv committed:
fix(themes): update component properties to use 'elevation' variables instead of 'shadow' - scoped (#16461)

* fix(themes): update properties to use 'elevation' variables instead of 'shadow' variables

* deps(theming): bump to the latest version

---------

Co-authored-by: Simeon Simeonoff <sim.simeonoff@gmail.com>

adrianptrv committed:
fix(tabs): replace igx-icon with igc-icon in WC demo + fix theme parameters (scoped) (#16415)

adrianptrv committed:
fix(tabs): set correct bootstrap focus border var (#16481)

didimmova committed:
fix(input): align input sass theme with wc (#16463)

SisIvanova committed:
refactor(radio): style component using sass theme (#16476)

adrianptrv committed:
fix(list): icon size in nested elements on initial load (#16498)

SisIvanova committed:
fix(switch): style component using sass theme (#16491)

* fix(switch): style component using sass theme

* fix(switch): indigo border-radius

* fix(switch): hover shadow

SisIvanova committed:
fix(combo): style component using sass theme (#16389)

SisIvanova committed:
refactor(themes): accessibility color fixes (#16518)

didimmova committed:
refactor(button): refactor button themes to match all repos (#16527)

SisIvanova committed:
refactor(grid): bootstrap color enhancement (#16546)

SisIvanova committed:
refactor(list): styling discrepancies

kdinev committed:
Merge pull request #16578 from IgniteUI/sivanova/list-styles

refactor(list): styling discrepancies

desig9stein committed:
[celendar] refactoring the component syles to match the master branch (#16600)

SisIvanova committed:
fix(*): styling discrepancies (#16604)

* fix(*): styling discrepancies

* fix(slider): remove flex-grow

---------

Co-authored-by: Simeon Simeonoff <sim.simeonoff@gmail.com>

adrianptrv committed:
fix(navbar): move icon padding to icon-button elements only (#16693)

adrianptrv committed:
refactor(chip): update chip states colors (#16691)

didimmova committed:
fix(excel-filtering): fix styles (#16707)

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

didimmova committed:
style(textarea): add padding bottom to textarea in all themes (#16808)

* style(textarea): add padding bottom to textarea in all themes

* fix(textare): align material textarea styles with master

didimmova committed:
fix(input-group): fix cursor behaviour in input groups (#16869)

* fix(input-group): fix cursor behaviour in input groups

* fix(input-group): fix cursor behaviour in simple-combo

* fix(combo): fix disabled toggle/clear button cursor

* chore(combo): remove additional indigo cursor styles on disabled

simeonoff committed:
Merge remote-tracking branch 'origin/master' into simeonoff/scoped-styles

simeonoff committed:
refactor(themes): change mechanism for scoped styles

simeonoff committed:
refactor(navdrawer): add missing styles

simeonoff committed:
refactor(themes): replace --igx prefixes with --ig

simeonoff committed:
refactor(navdrawer): streamline animation

simeonoff committed:
refactor(themes): bring back the grid theme

simeonoff committed:
docs: fix sassdoc for _standalone.scss

simeonoff committed:
refactor(themes): update standalone mixins

desig9stein committed:
Merge branch 'master' into simeonoff/scoped-styles

# Conflicts:
#	package-lock.json
#	src/app/app.component.html

desig9stein committed:
feat(input-group): pass hints QueryList from wrapper components to input-group

- Fix styling issues
- Wrapper components (combo, select, date-picker, time-picker,date-range-picker) now forward their @ContentChildren hints to the inner igx-input-group, matching the existing pattern for prefixes and suffixes. This ensures @if (hasHints) works correctly when hints are dynamically added or removed through wrapper components.

desig9stein committed:
fix(styles): update height properties to inherit for bundle components

desig9stein committed:
Merge remote-tracking branch 'origin/master' into simeonoff/scoped-styles

desig9stein committed:
fix(navigation-drawer): replace specificity placeholders with explicit styles and add active state theming

desig9stein committed:
fix(input-group): adjust hint row height and update padding; inherit height for bundled components

desig9stein committed:
fix(buttons): update icon sizing and alignment in button and icon-button themes

- Add `--size` variable for `igx-icon` within buttons for consistent sizing.
- Apply `aspect-ratio` to FAB buttons for proper circular appearance.

desig9stein committed:
fix(input-group): fix fluent inherit height for bundle elements

desig9stein committed:
fix(input-group): update input fluent styles to fix file height

desig9stein committed:
refactor(grid): scope styles to grid components (#16813)

simeonoff committed:
deps(theming): bump to the latest version

desig9stein committed:
feat(date-range-picker): fix styling issues

- fix issues in Material, Indigo, Fluent, and Bootstrap themes for the date-range-picker.
- Refactored date-range-picker grid layout for better alignment and flexibility.
- Adjusted input group size variables and spacings for consistency across themes.

desig9stein committed:
feat(action-strip): refactor sample layout and enhance styling

desig9stein committed:
fix(themes): update box-shadow and layout inconsistencies across themes

- Change `box-shadow` to `inset` for consistency in Material, Fluent, Bootstrap, and Indigo themes.
- Refactor date-range-picker sample to use dynamic labels and hints.
- Update theme variables to use Material palette/schema.
- Adjust grid-row layout for better alignment in date-range-picker.

desig9stein committed:
refactor(samples): remove local density chooser for each individual component in favor of global one,

 update layout styles for samples app.
 use igx-navbar.

desig9stein committed:
Merge remote-tracking branch 'origin/master' into simeonoff/scoped-styles

desig9stein committed:
Merge remote-tracking branch 'origin/master' into simeonoff/scoped-styles

# Conflicts:
#	projects/igniteui-angular/core/src/core/styles/components/grid/_grid-theme.scss

desig9stein committed:
refactor(samples): update sample styles for layout consistency

- Updated Divider, DropDown, and Properties Panel samples to use new layout configurations.
- Improved color sample functionality with hex copying and dynamic palette updates.
- Refactored Color sample to simplify structure and enhance usability.
- Applied scoped styles and updated component themes for consistency.
- Resolved layout inconsistencies in Divider and DropDown demos.

desig9stein committed:
fix(samples): ensure active item is scrolled into view

desig9stein committed:
Merge branch 'master' into simeonoff/scoped-styles

desig9stein committed:
refactor(themes, samples): improve styles and remove redundant functionality

- Refactored styling in Indigo, DropDown, Chips, and Grids samples
- Removed unused and duplicate properties in date-range-picker sample, simplifying configuration.
- Updated date-range-picker sample layout for a cleaner and more streamlined structure.

desig9stein committed:
refactor(samples): remove `@HostBinding` for dynamic size styles and simplify sample templates

desig9stein committed:
Merge branch 'master' into simeonoff/scoped-styles

desig9stein committed:
fix(themes): remove redundant styles and improve button-group alignment

- Removed `aspect-ratio: 1` from FAB button styles since it courses a visual bug in the samples repo where the buttons that have text + icon become to tall.
- Added `width: 100%` to button-group items for better container alignment if the buttongroup itself has display of inline block adn a hardcoded width.

desig9stein committed:
fix(themes): correct theme references and adjust styles for consistency

- Updated `themed-block` calls in date-range-picker themes to use correct theme references (Material and Fluent).
- Set `--aside-padding` for Fluent theme in demos settings.
- Refined styles in `pageHeading` component and updated `igx-input-group` to use `display: block grid`.
- Replaced `--_range-border-style` with `--_preview-border-style` in calendar days view styles.

desig9stein committed:
fix(date-range-picker): improve layout and grid alignment in date-range-picker

- Adjusted row and column templates for date-range-picker across themes for consistent alignment.
- Updated sample structure and styles to enhance readability and showcase use cases.
- Simplified SCSS variables and removed redundant styles in input-group and date-picker themes.
- Refined HTML and SCSS in samples for improved layout structure and maintainability.

desig9stein committed:
Merge remote-tracking branch 'origin/master' into simeonoff/scoped-styles

# Conflicts:
#	package.json
#	projects/igniteui-angular/core/src/core/styles/components/grid/_excel-filtering-theme.scss
#	projects/igniteui-angular/core/src/core/styles/components/grid/_grid-component.scss
#	projects/igniteui-angular/core/src/core/styles/components/grid/_header-row-component.scss

desig9stein committed:
refactor(chart-dialog): replace IgxDividerDirective with IgxDividerComponent for updated imports and module configuration

desig9stein committed:
chore(dependencies): update `igniteui-theming` to `v25.0.2`

desig9stein committed:
fix(themes): refine grid and filtering styles for consistency

- Removed redundant SCSS properties related to grid cell editing and header styles.
- Ensured consistent styling for filtered icons across themes.
- Added `background` property to filtering row for improved theme support.
- Consolidated and adjusted icon size definitions for Fluent, Material, Indigo, and Bootstrap themes.

desig9stein committed:
fix(themes): reorganize and clean up grid styles for consistency

- Restored misplaced grid row and cell styling blocks for improved structure.
- Corrected comment formatting and removed commented-out code for clarity

desig9stein committed:
refactor(themes, samples): clean up styles, improve structure, and update dependencies

- Removed unused `IgxIconComponent` imports and redundant SCSS styles for better maintainability.
- Updated sample layout in `grid-multi-row-layout` for improved readability and consistent structure.
- Adjusted `grid` and `properties-panel`

desig9stein committed:
chore(dependencies, themes): update `igniteui-theming` to `v25.1.0-beta.2` and switch to Material theme variables

- Updated `igniteui-theming` dependency to `v25.1.0-beta.2` in `package.json` and `package-lock.json`.
- Revert Indigo theme with Material theme

didimmova committed:
feat(checkbox): implement disabled-fill-color and apply to all themes

desig9stein committed:
Merge remote-tracking branch 'origin/master' into simeonoff/scoped-styles

# Conflicts:
#	package-lock.json
#	package.json
#	projects/igniteui-angular/combo/src/combo/combo.component.html
#	projects/igniteui-angular/simple-combo/src/simple-combo/simple-combo.component.html

desig9stein committed:
fix(scripts, dependencies): bring back concurrent tasks for styles watcher

desig9stein committed:
fix(themes): update filtered icon across themes

- Replaced `.icon--filtered` with `.igx-excel-filter__icon--filtered` in all themes.
- Added `display: inline-flex` style for filtered icon in base theme.
- Removed unnecessary `@extend` statement in `_base.scss`

desig9stein committed:
fix(input-group): prevent hints from flickering when value is changed externally
  When wrapper components (e.g. combo) project hints into the input-group,
  Angular's @ContentChildren re-evaluation during change detection would
  overwrite the hints set via the setter, causing the @if (hasHints)
  condition to momentarily evaluate to false and destroy the hint DOM.
  Separate @ContentChildren hints (_ownHints) from externally set hints
  (_externalHints) so they no longer conflict during change detection.

desig9stein committed:
fix(styles, samples): adjust drop-down and grid styles, add input placeholder

- Removed unused SCSS styling for `igx-input-group` in `drop-down.sample.scss`.
- Added `placeholder` attribute to input in `drop-down.sample.html`.
- Updated grid styles in `_base.scss` for consistency and simplified border styling.

desig9stein committed:
refactor(samples, styling): revamp grid-pdf-export sample layout and styling

- Replaced old SCSS structure with CSS Grid for modernized layout and consistency.
- Introduced property panel for dynamic PDF export configuration.
- Simplified and consolidated export functionality with `PropertyChangeService` integration.
- Enhanced toolbar structure for better usability across all grid types.

desig9stein committed:
refactor(themes): consolidate toolbar SCSS files, remove redundancies

- Merged and streamlined SCSS structure for grid toolbar themes.
- Removed redundant dark and light list/toolbar styling files.
- Adjusted `_base.scss` to use unified theme tokens for consistency.
- Cleaned up unused variables and improved maintainability.

desig9stein committed:
Merge remote-tracking branch 'origin/master' into simeonoff/scoped-styles

# Conflicts:
#	projects/igniteui-angular/core/src/core/styles/components/grid/_grid-theme.scss
#	projects/igniteui-angular/directives/src/directives/drag-drop/drag-drop.spec.ts

desig9stein committed:
fix(input-group): move styles to material them for lable positioning

- added `--_bundle-row` and `--_hint-row` variables for `.igx-input-group:has(:not(.igx-input-group__label))`.
- Ensured consistency across base and material theme SCSS files.

desig9stein committed:
refactor(samples, styles): simplify and modernize component samples layout

- Switched from custom flexbox layouts to responsive CSS Grid for cleaner structure and better alignment.
- Replaced old `items-wrapper` and related styles with streamlined `:host`-based grid layout.
- Unified button and form sample styles using updated variables and semantic class names.
- Integrated Ignite UI Material theme tokens for consistency.
- Adjusted SCSS imports for modularity and removed redundant encapsulation.

desig9stein committed:
chore(dependencies): update `igniteui-theming` to v25.1.0

desig9stein committed:
fix(styles): update navigation drawer and input group SCSS, improve samples

- Replaced `border` with `border-block` in navigation drawer popover.
- Added styles for input group `bundle-main` to improve input accessibility.
- Updated panel configuration labels and default values for better clarity.
- Added scroll bar color adjustments in the main navigation styling.

desig9stein committed:
fix(styles): update property label styling in properties panel

- Added color, font size, and line height to improve label readability and alignment with theme standards.

desig9stein committed:
fix(color-sample): add support for modern CSS color formats in hex conversion

- Replaced `rgbToHex` with `colorToHex` using a canvas for accurate conversion of all CSS color formats.
- Ensure `canvasCtx` is properly initialized and cleared for reliable handling.

desig9stein committed:
Merge branch 'master' into simeonoff/scoped-styles

desig9stein committed:
fix(checkbox-base-directive): update label class to include base CSS class for no mather the label possition

desig9stein committed:
fix(styles): add `--_bundle-row: 1` for all inputs inside the cell

desig9stein committed:
refactor(samples): Cleanup inputs sample

desig9stein committed:
fox(grid): use correct color variables for expand/collapse-all icon inside the header

desig9stein committed:
refactor(samples): set grid-template-rows for auto sizing

desig9stein committed:
refactor(samples): enhance layout and styling for improved usability

desig9stein committed:
fix(input-group): update grid row variables for input groups and labels

desig9stein committed:
Merge remote-tracking branch 'origin/master' into simeonoff/scoped-styles

# Conflicts:
#	projects/igniteui-angular/grids/grid/src/grid-filtering-ui.spec.ts

desig9stein committed:
fix(layout): wrap router outlet in a div for improved responsiveness

desig9stein committed:
Merge branch 'master' into simeonoff/scoped-styles

desig9stein committed:
fix(scombp): update igx-combo toggle button styles

desig9stein committed:
fix(simple-combo): add styles for simple combo component

desig9stein committed:
Merge branch 'master' into simeonoff/scoped-styles

desig9stein committed:
fix(styles): fix missing bracket

adrianptrv committed:
fix(dropdown): fix spacing for label items

didimmova committed:
feat(splitter): align splitter with the latest design handoff (#16997)

* feat(splitter): align splitter with the latest design handoff

* feat(splitter): use base-font-size var and a splitter-size var

didimmova committed:
refactor(button-group): refactor and update button group styles w/o overlay

didimmova committed:
chore(deps): update theming package

didimmova committed:
feat(query-builder): update fluent and/or button styles

didimmova committed:
feat(grid): fix ghost-row styles

didimmova committed:
fix(chips): update chips paddings and gaps

didimmova committed:
fix(badge): fix badge to work with content projection

didimmova committed:
fix(input-group): separate input and file-input tokens

adrianptrv committed:
feat(chips-area): add separate theme (#17205)

* feat(chips-area): add separate theme + grid sample and chips-area changes

* chore(): fix spacings

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
refactor: fix build errors after merging master

didimmova committed:
fix(elevations): fix indigo dark theme elevations

didimmova committed:
fix(date-range): fix discrepancies for daterange picker

simeonoff committed:
feat(grids): consolidate shared grid styles via adoptedStyleSheets

Replace four identical per-component stylesheets with a single shared
grid-base.styles.scss compiled to a TS string export. The base directive
registers it once into document.adoptedStyleSheets on first instantiation,
guarded by platform.isBrowser. All grid variants drop their redundant styleUrl.

Note: SSR renders without grid styles (no <style> tag injection); full SSR
support is a follow-up.

simeonoff committed:
fix(styles): rebuild grid-base.styles.ts on SCSS watch changes

simeonoff committed:
chore(scripts): clean up sass build scripts and suppress if-function deprecations

simeonoff committed:
refactor(scripts): streamline sass build and skip duplicate npm start build

- Clean up scripts/sass.mjs: drop dead code (compileSass,
  buildComponentStyles, buildBaseStyles, _postProcessor), consolidate
  shared sass config, use async mkdir, write external source maps for
  theme presets, throw instead of process.exit on theme failures.
- Move incremental rebuild logic into createStylesBuilder with a
  reverse + forward dep map for O(deps) rebuilds.
- watch-styles: use the builder for incremental rebuilds and accept
  --no-initial via scripts/get-args.mjs to skip the initial full build.
- package.json: pass --no-initial to watch:styles in `npm start` so
  styles aren't built twice on startup.

adrianptrv commented:
In the **Time Picker** and **Date Picker** components, after adding `encapsulation: ViewEncapsulation.None`, the component inline style `styles: [':host { display: block; }']` is no longer being applied. As a result, the components falls back to the default `display: inline` property.

This also breaks some of the samples. The issue is most noticeable in the **"Reactive Forms Overview"** sample.

[Component](https://github.com/user-attachments/assets/ac4c7087-09c4-4e20-97fc-9671382f1126)
[Sample](https://github.com/user-attachments/assets/b9140c22-75ca-4da1-b012-96ef1f274830)

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
fix: errors post igniteui-theming update

simeonoff committed:
deps: bump theming to 27.1.0

simeonoff committed:
npm: update package-lock.json

simeonoff committed:
chore(styles): remove deprecated component/theme files

These files and themes have been superseeded by the scoped styles for
each component.

simeonoff committed:
deps: update igniteui-sassdoc-theme

simeonoff committed:
npm: update package-lock.json

desig9stein committed:
fix(styles): update border styles for pinned rows and columns

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
fix: minor adjustments after merging master

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

didimmova committed:
feat(grid): update grid styles and theme builder sample (scoped-styles) (#17313)

* feat(grid): update grid styles and theme builder sample

* feat(grid): add updates for the grid and theme-builder sample

* chore(grid-sample): remove summaries patch background from controls

* feat(grid-sample): add hierarchical grid

* fix(grid): fix banner in grid styles

* feat(grid): add latest updates

* chore: merge upstream

---------

Co-authored-by: Simeon Simeonoff <sim.simeonoff@gmail.com>

simeonoff committed:
lint: fix errors

simeonoff committed:
build: increase build budget

simeonoff committed:
lint: fix stylelint warnings and errors

simeonoff committed:
fix(test): remove stale karma plugin require and unused test assets

- Remove require('@angular-devkit/build-angular/plugins/karma') from
  karma.non-grid, karma.hierarchical-grid, karma.pivot-grid,
  karma.test-perf and karma.tree-grid configs; the package is no longer
  installed and the framework string identifier is sufficient.
- Remove test-utils/assets from angular.json test assets; all references
  are string-only src attributes that never require the files to be
  served, eliminating intermittent EBADF errors on macOS 26 / Node 24.
- Fix month-picker spec: update 4 navigation tests still using old
  igx-calendar-view__item* selectors to igx-calendar-view-item*.
- Fix drop-down spec: CSS_CLASS_TABS updated to igx-tab-header; the
  aria-activedescendant test now queries the host element by tag instead
  of the inner list div whose class was moved by the style refactor.

simeonoff committed:
lint: fix errors

simeonoff committed:
spec(date-range-picker): update spec with the correct classes

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
lint: fix lint errors

simeonoff committed:
spec(combo): fix failing tests

simeonoff committed:
spec: fix drop-down, simple-combo, stepper, and tabs

simeonoff committed:
spec(query-builder): fix failing tests

simeonoff committed:
fix(styles): register component stylesheets once per document

IgxStylesRegistrar deduped registrations on a per-instance Map, but the
stylesheets it adds live on document.adoptedStyleSheets, which is shared
across every injector pointing at the document. Each additional injector
therefore re-adopted the same stylesheet, accumulating duplicate sheets.

Key the registry by Document so a stylesheet is registered only once per
document, regardless of how many injectors share it.

simeonoff committed:
fix(progress-bar): allow setting of two-tone gradient in circular bar

simeonoff committed:
spec(pivot-grid): stabilize filter chip dropdown tests

simeonoff committed:
spec(grid): fix some failing tests

simeonoff committed:
spec(grid): fix failing tests

simeonoff committed:
refactor(excel-style-search): bump small itemSize to 28px

simeonoff committed:
spec(colum-resizing): adjust tests to account for border

simeonoff committed:
fix(excel-style-filtering): cache search list height to avoid NG0100

simeonoff committed:
spec(utils): fix comparison utils

simeonoff committed:
spec(grid): re-baseline scroll/alignment tests for themed rendering

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
npm(package-lock): do a clean reinstall of all packages

simeonoff committed:
lint: fix linting errors

simeonoff committed:
spec(pdf-export): raise per-test timeout so slow exports don't leak

simeonoff committed:
fix(grid): validate primaryKey field once per primaryKey/data change

simeonoff committed:
fix(core): default formatDate to mediumDate when no format is supplied

simeonoff committed:
spec(grid-filtering): de-flake arrow-button chip scroll test

simeonoff committed:
spec(elements): make suite runnable in containers and de-flake projection tests

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

desig9stein committed:
fix(styles): fix grid filtering chips cut off

desig9stein committed:
fix(styles): fix pickers missing display on host

desig9stein committed:
fix(styles): fix material card missing border in elevated state

desig9stein committed:
feat(styles): add simple combo showcase component sample and fix simple combo styling bug

desig9stein committed:
feat(chat): enhance chat sample

desig9stein committed:
fix(styles): remove unnecessary host styles from multiple components

desig9stein committed:
feat(styles): add simple combo showcase component with Angular and WC examples

simeonoff committed:
refactor(themes): move component tokens to the global theme files

desig9stein committed:
fix(test): fix failing tests do to ricent changes

simeonoff committed:
Merge branch 'simeonoff/scoped-styles' of github.com:IgniteUI/igniteui-angular into simeonoff/scoped-styles

simeonoff committed:
refactor(themes): move chip tokens to global theme stylesheet

simeonoff committed:
refactor(checkbox): move the tokens to the global theme

simeonoff committed:
refactor(themes): move radio tokens to global theme

desig9stein committed:
feat(styles): make sure filtering chip inside the filtering cell has min width set to 0

simeonoff committed:
refactor(themes): add missing radio declaration from the generators

desig9stein committed:
feat(styles): atempt to fix the failing filtering cell test

desig9stein committed:
feat(styles): Revert changes for chips in grid filtering cells.

This has to be logged as separate issue

desig9stein committed:
fix(styles): correct padding variable usage in list item styles remove redundent list size in the sample panel

simeonoff committed:
refactor(themes): move input group tokens to the global theme files

desig9stein committed:
fix(icon-button): add cursor pointer

simeonoff committed:
refactor(themes): centralize component & derived tokens out of component bundles

simeonoff committed:
build(extras): add missing Sass compiler paths to ng-package.json

simeonoff committed:
refactor(grid): move component tokens to global theme file

simeonoff committed:
refactor(themes): remove unused code

simeonoff committed:
refactor(navdrawer): consume size tokens from theming schema (#17383)

* refactor(navdrawer): consume size tokens from theming schema

Source the drawer full/mini widths from the --ig-nav-drawer-size /
--ig-nav-drawer-size--mini schema tokens via var-get instead of
redeclaring them locally, aligning with the theming package.

Refs IgniteUI/igniteui-webcomponents#2240

* deps(theming): bump to latest version

desig9stein committed:
fix(date-range-picker): Apply grid only when we have both, range start and range end.

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
fix(grid): let filtering chips area shrink below content width on resize

simeonoff committed:
refactor(themes): change import paths

kdinev committed:
Merge branch 'master' into simeonoff/scoped-styles

kdinev committed:
chore(*): updating the bundle-test project

kdinev committed:
Merge branch 'simeonoff/scoped-styles' of https://github.com/IgniteUI/igniteui-angular into simeonoff/scoped-styles

simeonoff committed:
refactor(themes): switch component theming to runtime --ig-theme/--ig-theme-variant

- Rename _standalone.scss to _scoping.scss and move component-tokens
there, next to themed/themed-block — the single place that gates emitted
CSS(tokens and structural styles alike) behind the live --ig-theme and
--ig-theme-variant custom properties via @container style queries. This
lets components switch themes at runtime by flipping those two
properties instead of requiring a recompiled stylesheet per theme. Split
token emission from structural styles in button, icon-button, highlight,
label, rating, ripple, and tooltip so their tokens are gated the same
way as every other component's, without disturbing their existing
ig.base cascade layer for structural rules.

- Delete the now-dead combo/navdrawer/progress-bar theme mixins,
superseded by their components' own scoped themes/_base.scss files.

- Fix packaging: generators/_base.scss reaches
grids/paginator/query-builder's derived-theme partials via deep relative
paths, but ng-packagr never copied them into the published package,
breaking `theme()` for consumers building a custom theme.
build-theme-styles.mjs now copies them into dist and adjusts the
affected relative imports to match the packaged layout.

simeonoff committed:
lint: fix lint error in theme.token.ts

simeonoff committed:
fix(themes): restore $exclude support for scoped-styles components

simeonoff committed:
chore(migrations): add migrations for changes to mixins and components

adrianptrv committed:
feat(progress): fix values not animating on Safari (#17402)

simeonoff committed:
fix(input-group): add styling support for date/time/month/week input types

simeonoff committed:
fix(radio): don't animate checked/unchecked transitions on initial render

simeonoff committed:
fix(button-group): keep selected item above sibling hover/active/focus borders

simeonoff committed:
fix(badge): restore icon-only/dot zero-padding exemptions in indigo theme

simeonoff committed:
fix(input-group): set label transitions for border type to match textarea

simeonoff committed:
fix(input-group): use :has() instead of prefixed/suffixed classes in bootstrap

simeonoff committed:
lint: fix warnings in themes

desig9stein committed:
fix(styles): fix stepper issues

fix visualy hidden separator
fix firs step appearence when title position is start/end

simeonoff committed:
fix(date-picker): scope overlay shape to --dropdown/--dialog classes

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

desig9stein committed:
fix(styles): fix stepper issues

- fix visualy hidden separator
- fix firs step appearence when title position is start/end

desig9stein committed:
chore(styles): remove commented code

desig9stein committed:
fix(calendar): update date range styles

simeonoff committed:
refactor(grid): don't show overlay background on loading data

simeonoff committed:
Merge branch 'simeonoff/scoped-styles' of github.com:IgniteUI/igniteui-angular into simeonoff/scoped-styles

simeonoff committed:
fix(action-strip): don't leak icon-color into nested drop-down items

desig9stein committed:
fix(grid): add border to select all  checkbox

simeonoff committed:
fix(input): address input related issues

- Restore the missing Material line-type corner radius on the
Combo/Select toggle button and stretch consumer-supplied buttons to fill
the Fluent input's full height in prefix/suffix slots.
- Move themed()'s emission from the ig.components layer to ig.base

simeonoff committed:
fix(input-group): file input bundle height in fluent theme

simeonoff committed:
fix(stepper): remove active-only padding override causing horizontal override

simeonoff committed:
fix(grid): don't override compact excel-filter checkbox label typography

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

desig9stein committed:
fix(stepper): remove redundant flex-grow and top calculation from stepper styles

simeonoff committed:
fix(input-group): center date/time input value in Safari

simeonoff committed:
chore(demos): let input-group-showcase inherit size from global setting

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
fix(time-picker): remove duplicate box-shadow and stray background

simeonoff committed:
fix(icon,navbar): stop navbar icon-size override from leaking

simeonoff committed:
fix(input-group): scope Safari datetime centering fix to border variant

simeonoff committed:
fix(query-builder): don't override chip size in indigo

simeonoff committed:
fix(tabs): uppercase the item titles in indigo theme

simeonoff committed:
fix(grid): remove duplicate border on pivot grid row-dimension headers

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

desig9stein committed:
fix(styles): update theming to use dynamic layer names

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
fix(grids): fix cell-editing input misalignment in Indigo theme

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

desig9stein committed:
fix(stepper): correct hover and focus styles for separators in fluent and material themes

desig9stein committed:
fix(calendar): remove wrong style from fluent theme days view

desig9stein committed:
fix(grid): adjust expanderClass logic in hierarchical rows

simeonoff committed:
fix(grids): fix invalid cell-editing border height mismatch

simeonoff committed:
refactor(demos): overhaul overlay and overlay-presets samples

simeonoff committed:
refactor(demos): fix grid-theme-builder perf and split into components

simeonoff committed:
fix(themes): declare per-theme cascade layer in the global @layer order

Add ig.#{$theme} to the top-level `@layer` priority declaration so
theme-variant-specific styles (material/indigo/fluent/bootstrap) get an
explicit, fixed position between ig.typography and ig.derived, instead
of having their cascade priority depend on whichever stylesheet happens
to reference that layer name first.

simeonoff committed:
refactor(demos): rework grid-auto-size sample layout and controls

simeonoff committed:
fix(input-group): suppress label float transition on initial mount

simeonoff committed:
lint: fix errors

simeonoff committed:
fix(themes): match derived themes by host class, not tag name

desig9stein committed:
fix(grid): add border to pinned rows in scroll-start section

desig9stein committed:
fix(calendar): apply border style to month picker components in bootstrap theme

desig9stein committed:
fix(button-group): wrap button group content in a container

- the fox prevents curent display overrides on the group to break the layout

desig9stein committed:
fix(tabs): update tab header styles to use consistent spacing with gap property

desig9stein committed:
fix(button-group): move button iteration inside container

- ensures proper DOM structure by placing the button group iteration within a container
- adding an example of using `values` input for programmatically generated button groups in angular

desig9stein committed:
fix(grid): add border style to hierarchical empty hierarchical-expander cell

desig9stein committed:
fix(tabs): add gap between tab header items

when tabs header are templated igxSuffix/prefix go outside of the inner container, having no gap will make the user templated ites stack togather

desig9stein committed:
refactor(hierarchical-grid): adjust sample styles for header actions and grid layout.

desig9stein committed:
fix(list): add margin-start to align item actions

desig9stein committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
fix(button): emit icon size override in the theme layer

simeonoff committed:
fix(button): gate transitions behind a post-render ready modifier

simeonoff committed:
fix(icon-button): transitions, icon size, and fluent focus ring

simeonoff committed:
spec: fix some failing tests after refactoring

simeonoff committed:
test(grid): await deferred navigation render in aria-attributes test

simeonoff committed:
fix(migrations): remove dangling typography mixins and deep scss imports

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
chore: update hamburger menu icon to use icon button

simeonoff committed:
fix(demos): improve page header toggle accessibility and select sizing

simeonoff committed:
fix(input-group): add toggle-button spacing

simeonoff committed:
fix(pageHeading): center text in logo

simeonoff committed:
chore(demos): remove unused imports from pageHeading

simeonoff committed:
fix(input-group): align floating label timing with Material curve

simeonoff committed:
spec(month-picker): replace classes after merge with master

simeonoff committed:
chore: update CHANGELOG.md

simeonoff committed:
spec: add just-driven CI-parity Docker test runner

simeonoff committed:
chore: merge master into scoped-styles

simeonoff committed:
fix(drop-down): restore position absolute lost in scoped-styles merge

simeonoff committed:
fix(drop-down): shrink dropdown panel to content width

The '.igx-drop-down' panel had no intrinsic width, so once it lost the
container width constraint it previously inherited from the global
theme, it stretched to fill its nearest block ancestor.

'position: absolute' also restores shrink-to-fit sizing but pulls the
panel out of normal flow, which stops it from contributing its height
to '.igx-overlay__content' (whose height is otherwise never set
explicitly). That broke vertical-flip/positioning math in overlay
positioning strategies (regressed IgxAutocomplete top-direction
positioning and IgxSimpleCombo's open-upward-when-no-room behavior).

'width: max-content' gives the same shrink-to-fit sizing while keeping
the panel in normal flow, so it still drives the overlay content
wrapper's auto height correctly.

simeonoff committed:
Merge branch 'simeonoff/scoped-styles' of github.com:IgniteUI/igniteui-angular into simeonoff/scoped-styles

simeonoff committed:
fix(drop-down): shrink dropdown panel to content width

simeonoff committed:
chore: merge master branch

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
fix(grid): conditional scroll-start border for pinned/column-selection

simeonoff committed:
fix(icon): scope icon variables correctly and improve theme overrides

simeonoff committed:
fix(typography): native h1-h6/p overrides losing to global typography

Components rendering (or letting consumers project) native
heading/paragraph tags could no longer override the library's global
.ig-typography default, since that default became unlayered while
component overrides stayed in @layer base - unlayered always wins
regardless of specificity or layer order.

Mark affected elements with [data-no-typography] (via a template
attribute for static markup, or the new shared IgxNoTypographyDirective
composed through hostDirectives for consumer-projected content) so the
upstream igniteui-theming exclusion skips them entirely: navbar,
calendar, card, dialog, bottom-nav, expansion-panel, stepper, list,
tabs, navigation-drawer, grid-toolbar, excel-style-filtering,
column-actions, chips, snackbar, banner, tree-node, toast.

Also add an ig.reset cascade layer (lowest precedence, before ig.base)
so consumers can wrap third-party reset/normalize stylesheets in @layer
ig.reset and never have to fight them on specificity either.

simeonoff committed:
fix(drop-down): restore text-overflow ellipsis on item content

simeonoff committed:
refactor(typography): exclude global h1-h6 and p styles for marked components

simeonoff committed:
deps(theming): bump to latest version

simeonoff committed:
fix(theming): bump igniteui-theming to 27.5.1 and fix base styles

simeonoff committed:
fix(theming): emit scoped tokens outside themed() wrapper for global components and directives

ChronosSF committed:
Merge branch 'master' into simeonoff/scoped-styles

simeonoff committed:
style(*): use mixin for shared input/textarea styles, trim badge text

Replace the %input-textarea-common placeholder selector with an
input-display() mixin in input-group to fix specificity issues, reused
by both @include e(input) and @include e(textarea).

Add text-box: trim-both cap alphabetic to badge value for tighter
vertical text metrics.

desig9stein committed:
fix(calendar): fix tailwind reset breaking the range selection

simeonoff committed:
fix(card): header doesn't use global typography

simeonoff committed:
fix(grid): exclude bool columns from fixed-width td cell styles

Add `$not: bool` modifier exclusion to the fixed-width, non-resizable td
selector so boolean column cells no longer pick up the `fw` modifier
styles meant only for other fixed-width columns.

simeonoff committed:
fix(query-builder): scope flat button CSS custom properties

Prefix the and/or expression-group flat button color custom properties
with ig-flat-button- so they target the scoped button theme variables
instead of the previous unscoped names that no longer applied after the
button theme's variable scoping change.

desig9stein committed:
fix(calendar): remove igx-calendar-picker--standalone token scope

  The scope redeclared the full calendar token set on the picker element
  inside igx-month-picker, shadowing any override set on the host.

simeonoff committed:
fix(button): preserve layered styles on directive hosts

simeonoff committed:
fix(pivot-data-selector): remove block margin from section titles

simeonoff committed:
chore(demos): stretch pivot sample to full height

desig9stein committed:
fix(grid): remove `overflow: hidden` from body cells to fix drag selection issue on Firefox

simeonoff committed:
chore(changelog): guide users on how to properly migrate to the tokens mixin

desig9stein committed:
fix(input-group): account for `hasBadInput` in `isFilled` calculation

- adds `hasBadInput` check to ensure proper handling of invalid user input in `type="number"` and similar inputs.

desig9stein committed:
fix(gird): adjust styles for input components inside grid cell

desig9stein committed:
fix(grid): refine styles for input components in editable cells

desig9stein committed:
feat(grid): add cascading combos sample with dynamic region and city updates

- Adds a new sample demonstrating cascading combo functionality within a grid.
- Includes dynamic data updates for regions and cities based on country selection.
- Provides a visual progress indicator for async data loading.

simeonoff committed:
fix(grid): display pointer cursor on filtering icon
